### ISSUE TITLE
angular momentum conservation

### ISSUE THREAD

**Issue by migueldvb** [piernik-dev/piernik#200]
Do you know if there are any tests for the conservation of angular momentum in Piernik's Cartesian geometry implementation using AMR?

**Comment by gawrysz** [comment]
No, I'm afraid we haven't been designing such conservation tests on cartesian grid. I think problems/wt4 would be a good starting point.

Major obstacle is that for cartesian geometry with open boundaries we don't have much control on leaks/inflow of angular momentum. Perhaps the best that can be done would be to integrate angular momentum fluxes (some other fluxes as well) on external boundaries and use them in .tsl files to give an idea what's going on.

Soon we will have MHD Riemann solver in main branch of Piernik which should be way less diffusive than RTVD. Making it fully usable with AMR may take some time though. Contributions are welcome :)

**Comment by migueldvb** [comment]
Thank you @gawrysz . The Cartesian geometry disk problem will help.

**Comment by Xarthisius** [comment]
The conservation of angular momentum was/is poor even on a uniform Cartesian grid. That's why I had to implement cylindrical coordinates for my thesis. I don't think it's going to be better with AMR.
